src/client/mesh_generator_thread.cpp

```diff
@@ -146,16 +146,25 @@ QueuedMeshUpdate *MeshUpdateQueue::pop()
 	for (std::vector<QueuedMeshUpdate*>::iterator i = m_queue.begin();
 			i != m_queue.end(); ++i) {
 		QueuedMeshUpdate *q = *i;
-		if(must_be_urgent && m_urgents.count(q->p) == 0)
+		if (must_be_urgent && m_urgents.count(q->p) == 0)
+			continue;
+		if (m_inflight_blocks.find(q->p) != m_inflight_blocks.end())
```

Comment:
This is to avoid multiple threads putting the same block on the queue?
Might want to add a comment.

Reply:
This is to prevent a block being put into the queue again if it's currently being processed.

Reply:
This is to prevent the same block from being processed by two threads concurrently, as that causes racing conditions and wastes resources. Comment added.